### PR TITLE
Refactored preprocessing

### PR DESCRIPTION
**Pull request by alexteua**
Using torchaudio and batched processing where possible
General cleanup

### CODE REVIEW

File: utils/dsp.py [hunk 1]
```diff
@@ -47,35 +51,92 @@ def __init__(self,
...
+        return list_of_mels
+
+    def waveform_to_mel(self, waveform: torch.Tensor, normalized=True) -> torch.Tensor:
+        """Convert waveform to mel spectrogram"""
+        mel_spec = self.mel_transform(waveform)
+        if normalized:
+            mel_spec = self.normalize(mel_spec)
+        return mel_spec
 
     def griffinlim(self, mel: np.array, n_iter=32) -> np.array:
```
Comment (cschaefer26) [inline utils/dsp.py:141]: I would suggest to use the same type (torch.Tensor) for all functions to be consistent (now some run on np.arrays and some on tensors)

**Review by cschaefer26 — CHANGES_REQUESTED** [review]
- torchaudio missing in requirements.txt
- tests not passing
- could be useful to add a test for some of the changed audio processing, e.g testing if the batched processing yields the exact same results as non-batched
- bug in preprocessing (voice_encoder running on wrong input)

File: configs/multispeaker.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ dsp:
   win_length: 1024
   fmin: 0
   fmax: 8000
+  target_dBFS: -30                      # Target loudness in decibels, used for normalization
```
[the remark this answers is not in the document]
Comment (alexteua) [reply configs/multispeaker.yaml:17]: @cschaefer26 it's for the reference wav processing, and yes it's always applied, the same as it was in the old preprocessing

**Review by cschaefer26 — APPROVED** [review]

File: preprocess.py [hunk 3]
```diff
@@ -156,35 +163,49 @@ def _convert_file(self, item_id: str, path: Path) -> DataPoint:
...
-                cleaned_texts += [(dp.item_id, dp.text)]
-                successful_ids.add(dp.item_id)
-            except Exception as e:
-                print(traceback.format_exc())
+    # run processing
+    start_time = time.monotonic()
+
+    # Prepare processing
+    tts_dataset = TTSDataset(file_id_audio_list)
+    dataloader = DataLoader(tts_dataset,
```
[the remark this answers is not in the document]
Comment (alexteua) [reply preprocess.py:177]: They are sorted by length in the TTSDataset, and then shuffle is set to False for the Dataloader, so the batches are created on a sorted data.
